include/deal.II/base/exceptions.h

```diff
@@ -1519,6 +1519,49 @@ namespace deal_II_exceptions
 } /*namespace deal_II_exceptions*/
 
 
+#if defined(__clang__)
+#  define DEAL_II_ASSUME(expr)                     \
+    do                                             \
+      {                                            \
+        __builtin_assume(static_cast<bool>(expr)); \
+      }                                            \
+    while (false)
```

Comment:
Since you have a single statement, you don't actually need the `do ... while(false)` part around it.